components/dashboard/src/contexts/FeatureFlagContext.tsx

```diff
@@ -4,19 +4,46 @@
  * See License-AGPL.txt in the project root for license information.
  */
 
-import React, { createContext, useContext, useEffect } from "react";
+import React, { createContext, useContext, useState, useEffect } from "react";
+import { useLocation } from "react-router";
+import { getExperimentsClient } from "../experiments/client";
+import { ProjectContext } from "../projects/project-context";
+import { getCurrentTeam, TeamsContext } from "../teams/teams-context";
 import { UserContext } from "../user-context";
 
-const FeatureFlagContext = createContext({});
+const FeatureFlagContext = createContext<{ showUsageBasedPricingUI: boolean }>({
+    showUsageBasedPricingUI: false,
+});
 
 const FeatureFlagContextProvider: React.FC = ({ children }) => {
```

Comment:
Would this Provider now need to implement every feature flag?

My thinking was that the component would accept (flagName, params) and return the value without needing to explicitly implement each flag in the ContextProvider

Reply:
The context is used here: 

https://github.com/gitpod-io/gitpod/blob/655eca7d3a2a867299b863bc057d51a99ab981bd/components/dashboard/src/index.tsx#L34-L36

ie at the very top level of the component tree, so it can't usefully take props. The idea of the context is that it gets the value of all feature flags and makes them available so that components further down the tree can look up the value of a flag like this:

https://github.com/gitpod-io/gitpod/blob/655eca7d3a2a867299b863bc057d51a99ab981bd/components/dashboard/src/settings/PageWithSettingsSubMenu.tsx#L20

The advantage is consolidation of all the feature flag retrieval logic in one place (in the context).

Reply:
That's fair, just may not be intuitive for users that they also need to go implement the fetching. I guess it feels like a bit too much work to use a feature flag but I get it, and perhaps there isn't a better pattern in React